interfaces/Portalicious/src/app/services/auth/strategies/msal-auth/msal-auth.strategy.ts

```diff
@@ -39,13 +43,25 @@ export class MsalAuthStrategy implements IAuthStrategy {
   }
 
   public async login(credentials: { username: string }) {
+    const loginRequest: PopupRequest | RedirectRequest = {
+      scopes: [`api://${environment.azure_ad_client_id}/User.read`],
+      loginHint: credentials.username,
+    };
+
     if (!isIframed()) {
-      this.msalService.loginRedirect({
-        scopes: [`api://${environment.azure_ad_client_id}/User.read`],
-        loginHint: credentials.username,
-      });
+      this.msalService.loginRedirect(loginRequest);
     } else {
-      throw new Error('TODO: AB#31469 Implement loginPopup for iframe');
+      const sub = this.msalService
+        .loginPopup(loginRequest)
+        .subscribe((data: AuthenticationResult | null) => {
+          sub.unsubscribe();
+          if (!data) {
+            throw new Error(
+              'MSAL Strategy: an error occurred while logging in with popup',
+            );
+          }
+          void this.router.navigate(['/', AppRoutes.authCallback]);
+        });
     }
 
     // The user is being fetched & set in msal-auth.callback.component
```

Comment:
Does this also apply to the popup logic?

Reply:
I'm not sure I understand your question.
The router.navigate is needed in the popup subscribe because in this scenario we don't have a redirectUrl (like in loginRedirect). We have to handle the 'redirects' ourself by navigating to the page.

Reply:
Ah ok. Then I think the current code is a bit misleading, but I'll add a suggestion to clarify.